src/plugins/mer/meremulatordevice.cpp

```diff
@@ -255,6 +261,18 @@ void MerEmulatorDevice::executeAction(Core::Id actionId, QWidget *parent)
         return;
     } else if (actionId == Constants::MER_EMULATOR_STOP_ACTION_ID) {
         m_connection->disconnectFrom();
+        return;
+    } else if (actionId == Constants::MER_EMULATOR_CHANGE_MODE_ACTION_ID) {
+        const IDevice::ConstPtr device = DeviceManager::instance()->find(id());
+        QTC_ASSERT(device, return);
```

Comment:
When new emulator is being added, `device` will be null until the changes are applied. In this case it might be OK to work on `this` instance instead.

If you want to test adding an emulator, you need to manually edit devices.xml, change "Origin" to "0", then you the "Remove" button will become enabled, so that you can remove the emulator and add it later.

Reply:
Fixed. Using copy in this case.